File: Lab15_1/UniversityUnit.cs
using System;

namespace Lab15_1
{
    public class UniversityUnit:IComparable<UniversityUnit>
    {
        public string UnitName { get; set; }
        public string UniversityName { get; set; }
        public int Employees { get; set; }

        public UniversityUnit(string data)
        {
            string[] dataArray = data.Split(',');
            this.UnitName = dataArray[0];
            this.UniversityName = dataArray[1];
            this.Employees = Convert.ToInt32(dataArray[2]);
        }
        
        
        public UniversityUnit(string unitName, string universityName, int employees)
        {
            UnitName = unitName;
            UniversityName = universityName;
            Employees = employees;
        }
        
        public int CompareTo(UniversityUnit other)
        {
            if (other == null)
                return 1;
            return UnitName.CompareTo(other.UnitName);
        }
        
        
        
        public override string ToString()
        {
            return $"{UnitName}, {UniversityName}, {Employees}";
        }

        public override bool Equals(object obj)
        {
            var cmp = (UniversityUnit)obj;
            if (this.UnitName == cmp.UnitName && this.UniversityName == cmp.UniversityName && this.Employees == cmp.Employees)
                return true;
            return false;
        }
        
        
    }
}

File: python/task1/UniversityUnit.py
class UniversityUnit:

    def __init__(self, data):
        dataArray = data.split(',')
        name = dataArray[0]
        universityName = dataArray[1]
        employees = dataArray[2]
        self.__name = name
        self.__universityName = universityName
        self.__employees = int(employees)

    def get_name(self):
        return self.__name

    def get_university_name(self):
        return self.__universityName

    def get_employees(self):
        return self.__employees

    def set_name(self, name):
        self.__name = name

    def set_university_name(self, universityName):
        self.__universityName = universityName

    def set_employees(self, employees):
        self.__employees = int(employees)

    def __str__(self):
        return f"{self.__name}, {self.__universityName}, {self.__employees}"

    def __eq__(self, other):
        if self.get_name() == other.get_name() and self.get_university_name() == other.get_university_name() and self.get_employees() == other.get_employees():
            return True
        return False

    def __lt__(self, other):
        if self.get_name() < other.get_name():
            return True
        return False